RevenueCatUI/Templates/V2/Components/Button/ButtonComponentView.swift

```diff
@@ -142,7 +146,17 @@ struct ButtonComponentView: View {
         }
     }
 
-    private func openCodeRedemptionSheet() {
+    private func openCodeRedemptionSheet() async {
+        // Check if there's an offer code redemption interceptor
+        if let interceptor = self.offerCodeRedemptionInitiatedAction {
+            // Wait for the interceptor to call resume before proceeding
+            await withCheckedContinuation { continuation in
+                interceptor(resume: {
+                    continuation.resume()
+                })
+            }
```

Comment:
Same comment here about having `actionTypeInProgress` return true to block the UI while the continuation happens